src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamPackageSpecReader.cs

```diff
@@ -0,0 +1,1873 @@
+            if (!wasPackOptionsSet)
+            {
+                packageSpec.Owners = Array.Empty<string>();
+                packageSpec.PackOptions = new PackOptions()
+                {
+                    PackageType = Array.Empty<PackageType>()
+                };
+                packageSpec.Tags = Array.Empty<string>();
+            }
+
+            if (isMappingsNull)
+            {
+                packageSpec.PackOptions.Mappings = null;
+            }
+#pragma warning restore CS0612 // Type or member is obsolete
+
+            packageSpec.RuntimeGraph = new RuntimeGraph(
+                runtimeDescriptions ?? Enumerable.Empty<RuntimeDescription>(),
+                compatibilityProfiles ?? Enumerable.Empty<CompatibilityProfile>());
+
+            if (packageSpec.Name == null)
+            {
+                packageSpec.Name = packageSpec.RestoreMetadata?.ProjectName;
+            }
+
+            // Use the project.json path if one is set, otherwise use the project path
+            if (packageSpec.FilePath == null)
+            {
+                packageSpec.FilePath = packageSpec.RestoreMetadata?.ProjectJsonPath
+                    ?? packageSpec.RestoreMetadata?.ProjectPath;
+            }
+
+            return packageSpec;
+        }
+
+        internal static void ReadCentralTransitiveDependencyGroup(
+            ref Utf8JsonStreamReader jsonReader,
+            IList<LibraryDependency> results,
+            string packageSpecPath)
+        {
+            if (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.StartObject)
+            {
+                while (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.PropertyName)
+                {
+                    var propertyName = jsonReader.GetString();
+                    if (string.IsNullOrEmpty(propertyName))
+                    {
+                        throw FileFormatException.Create(
+                            "Unable to resolve dependency ''.",
+                            packageSpecPath);
+                    }
+
+                    if (jsonReader.Read())
+                    {
+                        var dependencyIncludeFlagsValue = LibraryIncludeFlags.All;
+                        var dependencyExcludeFlagsValue = LibraryIncludeFlags.None;
+                        var suppressParentFlagsValue = LibraryIncludeFlagUtils.DefaultSuppressParent;
+                        string dependencyVersionValue = null;
+
+                        if (jsonReader.TokenType == JsonTokenType.String)
+                        {
+                            dependencyVersionValue = jsonReader.GetString();
+                        }
+                        else if (jsonReader.TokenType == JsonTokenType.StartObject)
+                        {
+                            while (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.PropertyName)
+                            {
+                                IEnumerable<string> values = null;
```

Comment:
you probably don't need to declare this variable. You just assign and use it within the condition.